### PR TITLE
Refactoring of GTR template checker.

### PR DESCRIPTION
**Pull request by peria**
Line number 41 and 67 in the original code had wrong indecies.

### CODE REVIEW

File: src/cpu/test_lockit/template.cc [hunk 1]
```diff
@@ -4,99 +4,43 @@
...
-
-    return sc;
+    int score = 0;
+
+    int expect_same[][4] = {
+        {0, 0, 1, 0}, {0, 0, 1, 2}, {0, 0, 2, 1},
+        {1, 0, 1, 2}, {1, 0, 2, 1}, {1, 2, 2, 1},
+        {0, 1, 0, 2}, {0, 1, 1, 1}, {1, 1, 0, 2},
+    };
+    for (auto& xy : expect_same) {
```
Comment (mayah) [inline src/cpu/test_lockit/template.cc]: const auto&

File: src/cpu/test_lockit/template.cc [hunk 2]
```diff
@@ -4,99 +4,43 @@
...
-    if ((f[2][1] != TLColor::EMPTY) && (f[2][2] != TLColor::EMPTY)) {
-        if (f[2][1] == f[2][2]) {
-            sc -= 500;
-        }
-    }
-
-    return sc;
+    int score = 0;
+
+    int expect_same[][4] = {
```
Comment (mayah) [inline src/cpu/test_lockit/template.cc]: static const int

File: src/cpu/test_lockit/template.cc [hunk 3]
```diff
@@ -4,99 +4,43 @@
...
+        int x1 = xy[2], y1 = xy[3];
+        if (field[x0][y0] == TLColor::EMPTY || field[x1][y1] == TLColor::EMPTY)
+            continue;
+        if (field[x0][y0] == field[x1][y1])
+            score += 1000;
+        else
+            score -= 1000;
+    }
+
+    int expect_diff[][5] = {
```
Comment (mayah) [inline src/cpu/test_lockit/template.cc]: static const int

File: src/cpu/test_lockit/template.cc [hunk 4]
```diff
@@ -4,99 +4,43 @@
...
+        {0, 2, 0, 3, 2000},
+        {1, 2, 1, 3, 500}, {1, 2, 2, 2, 500}, {2, 1, 2, 2, 500},
+        {1, 0, 2, 0, 1000}, {2, 1, 2, 0, 1000},
+    };
+    for (auto& xy : expect_diff) {
```
Comment (mayah) [inline src/cpu/test_lockit/template.cc]: const auto&

File: src/cpu/test_lockit/template.cc (hunk 2 above)
Comment (mayah) [inline src/cpu/test_lockit/template.cc]: How about
````
static const struct {
    int x0, y0, x1, y1;
} expect_same[] = {
   { ... },
   { ... },
};
````

File: src/cpu/test_lockit/template.cc (hunk 1 above)
Comment (peria) [inline src/cpu/test_lockit/template.cc]: done

File: src/cpu/test_lockit/template.cc (hunk 2 above)
Comment (peria) [inline src/cpu/test_lockit/template.cc]: done

File: src/cpu/test_lockit/template.cc (hunk 3 above)
Comment (peria) [inline src/cpu/test_lockit/template.cc]: done

File: src/cpu/test_lockit/template.cc (hunk 4 above)
Comment (peria) [inline src/cpu/test_lockit/template.cc]: done

**Comment by mayah** [comment]
[LGTM](https://pbs.twimg.com/media/CJzqdhPVAAAKqGz.png:large)

**Pull request merged**